src/runtime/typeoffset.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Reflection;
+
+namespace Python.Runtime
+{
+    internal static partial class TypeOffset
```

Comment:
Other offset types still in the `interop.cs`, why just extracts it alone. I think it's not a necessary change for shutdown mode features, it only increasing the changes for merging. Maybe you can bring these on upstream at first.

Reply:
Good point, I'll revert that